components/style/properties/longhand/color.mako.rs

```diff
@@ -53,4 +55,91 @@
+
+            # These are not parsed but must be serialized
+            # They are only ever set directly by Gecko
+            extra_colors = """WindowBackground WindowForeground WidgetBackground WidgetForeground
+                              WidgetSelectBackground WidgetSelectForeground Widget3DHighlight Widget3DShadow
+                              TextBackground TextForeground TextSelectBackground TextSelectForeground
+                              TextSelectForegroundCustom TextSelectBackgroundDisabled TextSelectBackgroundAttention
+                              TextHighlightBackground TextHighlightForeground IMERawInputBackground
+                              IMERawInputForeground IMERawInputUnderline IMESelectedRawTextBackground
+                              IMESelectedRawTextForeground IMESelectedRawTextUnderline
+                              IMEConvertedTextBackground IMEConvertedTextForeground IMEConvertedTextUnderline
+                              IMESelectedConvertedTextBackground IMESelectedConvertedTextForeground
+                              IMESelectedConvertedTextUnderline SpellCheckerUnderline""".split()
+        %>
+        use gecko_bindings::bindings::Gecko_GetLookAndFeelSystemColor;
+        use gecko_bindings::structs::root::mozilla::LookAndFeel_ColorID;
+        pub type SystemColor = LookAndFeel_ColorID;
+
+        impl ToCss for SystemColor {
+            fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+                let s = match *self {
+                    % for color in system_colors + extra_colors:
+                        LookAndFeel_ColorID::eColorID_${to_rust_ident(color)} => "${color}",
+                    % endfor
+                    LookAndFeel_ColorID::eColorID_LAST_COLOR => unreachable!(),
+                };
+                dest.write_str(s)
+            }
+        }
+
+        impl ToComputedValue for SystemColor {
+            type ComputedValue = u32; // nscolor
+            #[inline]
+            fn to_computed_value(&self, cx: &Context) -> Self::ComputedValue {
+                unsafe {
+                    Gecko_GetLookAndFeelSystemColor(*self as i32,
+                                                    &*cx.device.pres_context)
+                }
+            }
+
+            #[inline]
+            fn from_computed_value(_: &Self::ComputedValue) -> Self {
+                unreachable!()
+            }
+        }
+
+        impl SystemColor {
+            pub fn parse(input: &mut Parser) -> Result<Self, ()> {
+                use std::ascii::AsciiExt;
+                static PARSE_ARRAY: &'static [(&'static str, SystemColor); ${len(system_colors)}] = &[
+                    % for color in system_colors:
+                        ("${color}", LookAndFeel_ColorID::eColorID_${to_rust_ident(color)}),
+                    % endfor
+                ];
+
+                let ident = input.expect_ident()?;
+                for &(name, color) in PARSE_ARRAY.iter() {
```

Comment:
I believe it's worth to be consistent. And `case_insensitive_phf_map` shouldn't be slower or anything.